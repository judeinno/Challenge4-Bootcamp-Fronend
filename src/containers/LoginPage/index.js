import React, { Component } from 'react';
import * as PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { loginData, loginThunk } from '../../actions/loginActions';
import Login from '../../components/login';

class LoginPage extends Component {
  handleChange = (event) => {
    event.preventDefault();
    const { loginUserData, loginReducer } = this.props;
    loginUserData({
      ...loginReducer.loginPostData,
      [event.target.name]: event.target.value,
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const { loginUser, loginReducer } = this.props;
    loginUser(loginReducer.loginPostData);
  }

  render() {
    return (
      <Login
        onChange={this.handleChange}
        onClick={this.handleSubmit}
      />
    );
  }
}

LoginPage.propTypes = {
  loginUser: PropTypes.func.isRequired,
  loginUserData: PropTypes.func.isRequired,
  loginReducer: PropTypes.shape({}).isRequired,
};

const mapStateToProps = ({ loginReducer }) => ({
  loginReducer,
});

export const mapDispatchToProps = dispatch => ({
  loginUser: response => dispatch(loginThunk(response)),
  loginUserData: response => dispatch(loginData(response)),
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
