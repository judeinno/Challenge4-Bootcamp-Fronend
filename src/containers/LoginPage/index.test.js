import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import LoginPage, { mapDispatchToProps } from './index';

describe('<LoginPage />', () => {
  let LoginPageComponent;
  beforeEach(() => {
    jest.resetModules();
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);
    const initialState = { loginReducer: { loginPostData: { username: 'Jude', Password: 'password' } } };
    const store = mockStore(initialState);
    LoginPageComponent = mount(
      <LoginPage store={store} />,
    );
  });

  it('should render the component', () => {
    expect(LoginPageComponent).toMatchSnapshot();
  });
  const dispatchTest = dispatch => (
    mapDispatchToProps(dispatch).loginUser({})
    && mapDispatchToProps(dispatch).loginUserData({})
  );
  it('should dispatch a method to get user input', () => {
    const dispatch = jest.fn();
    dispatchTest(dispatch);
  });

  it('should test user data without fail', () => {
    LoginPageComponent
      .find('#user-name')
      .simulate('change', { target: { name: 'username', value: 'Jude' } });
    LoginPageComponent
      .find('#sign-up-bt')
      .simulate('click');
  });
});
