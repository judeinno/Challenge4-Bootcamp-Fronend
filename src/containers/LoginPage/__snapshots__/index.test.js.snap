// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should render the component 1`] = `
<Connect(LoginPage)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LoginPage
    loginReducer={
      Object {
        "loginPostData": Object {
          "Password": "password",
          "username": "Jude",
        },
      }
    }
    loginUser={[Function]}
    loginUserData={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <Login
      onChange={[Function]}
      onClick={[Function]}
    >
      <div
        className="container signup"
      >
        <form
          className="signup-form"
        >
          <legend>
            Register
          </legend>
          <hr />
          <div
            className="form-group"
          >
            <p>
              Username:
            </p>
            <input
              className="form-control"
              id="user-name"
              name="username"
              onChange={[Function]}
              placeholder="User Name"
              type="test"
            />
          </div>
          <div
            className="form-group"
          >
            <p>
              Password:
            </p>
            <input
              className="form-control"
              id="exampleInputPassword1"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <button
            className="btn btn-primary"
            id="sign-up-bt"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </Login>
  </LoginPage>
</Connect(LoginPage)>
`;
